src/main/java/com/agnieszkapawska/flashcards/domain/services/QuestionTagService.java

```diff
@@ -0,0 +1,44 @@
+package com.agnieszkapawska.flashcards.domain.services;
+
+import com.agnieszkapawska.flashcards.domain.models.Flashcard;
+import com.agnieszkapawska.flashcards.domain.models.QuestionTag;
+import com.agnieszkapawska.flashcards.domain.repositories.QuestionTagRepository;
+import lombok.AllArgsConstructor;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@AllArgsConstructor
+public class QuestionTagService {
+    private QuestionTagRepository questionTagRepository;
+
+    public Set<QuestionTag> getQuestionTagsSet(Set<String> questionTagsName, Flashcard flashcard) {
+        Set<QuestionTag> questionTagSet = new HashSet<>();
+        for (String questionTagName:questionTagsName){
+            QuestionTag questionTag = checkQuestionTagIsPresentIfNotCreateNewQuestionTagAndAddFlashcardToItsSet(questionTagName, flashcard);
+            questionTagSet.add(questionTag);
+        }
+        return questionTagSet;
+    }
+
+    private QuestionTag checkQuestionTagIsPresentIfNotCreateNewQuestionTagAndAddFlashcardToItsSet(String questionTagName, Flashcard flashcard) {
+        Optional<QuestionTag> optionalQuestionTag = checkQuestionTagIsPresentByName(questionTagName);
```

Comment:
Assuming we don't want to add flashcard to set here, we could simplify this method to this (we also rename `checkQuestionTagIsPresentByName` to `getExistingTag`):
```
return getExistingTag(name).orElseGet {
    QuestionTag tag = new QuestionTag(name);
    tag.save(questionTag);
    return tag;
}
```